issuetracker/gitlab/gitlab.go

```diff
@@ -33,14 +34,19 @@ func (it *IssueTracker) taskURLFrom(taskID string) string {
 	return taskID
 }
 
-// IssueAPIOrigin returns the URL for github's issue-fetching API
+// IssueAPIOrigin returns the URL for gitlab's issue-fetching API
 func (it *IssueTracker) issueAPIOrigin() string {
-	tokens := common.RemoveEmptyTokens(strings.Split(it.Origin, "/"))
-	if !strings.HasPrefix(tokens[0], "http:") && !strings.HasPrefix(tokens[0], "https:") {
-		tokens = append([]string{"https:"}, tokens...)
+
+	pattern := regexp.MustCompile(`^(?P<scheme>^(https?:))//(?P<domain>gitlab.([a-zA-Z0-9\-]+)?(.)?com)/(?P<owner>[a-zA-Z0-9\-]+)/(?P<repo>[a-zA-Z0-9-_.]+)/?$`)
```

Comment:
gitlab can be self-hosted. It should be able to match any host e.g. `myorg.issuetracker.com`

Reply:
In this case, you expect someone to explicitly have `gitlab` in the hostname which can not always be the case.